src/zabapgit_http.prog.abap

```diff
@@ -494,6 +499,51 @@ CLASS lcl_http IMPLEMENTATION.
                                            iv_username = lv_user ).
     ENDIF.
 
+    " Is the repository hoster supported for using two factor authentication?
+    TRY.
+        IF lcl_2fa_authenticator_registry=>is_url_supported( iv_url ) = abap_true.
```

Comment:
can we take most of this part and move to a method in the 2fa include? and have it raise lcx_exception